src/context/useInstance.ts

```diff
@@ -0,0 +1,8 @@
+import { DocumentViewerContext } from './Context'
+import { useContext } from 'react';
+import { TContextState } from './Context';
+
+export default function useInstance(): TContextState{
+  const { instance, setInstance } = useContext(DocumentViewerContext);
+  return { instance, setInstance }
```

Comment:
This should be memoized

Reply:
Or you can just do `return useContext(DocumentViewerContext);`, then you don't need to memoize

Reply:
fixed